### PR TITLE
`Communication`: Migrate exercise/lecture posts to messages

### PR DESCRIPTION
**Pull request by lennart-keller**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

With #6548, exercises and lectures support channel messages as well of course posts to be displayed on the exercise/lecture pages if communication or messaging is enabled. 

### Description

This PR introduces a migration script, that moves lecture and exercise posts from courses with both communication and messaging enabled to the new system using channels. It creates the needed lecture/exercise channels and adds all course members as participants. Exercise/Lecture posts are moved to the belonging channels.

### Steps for Testing

Prerequisites:
- 1 Instructor
- A course with exercise/lecture posts and communication/messaging enabled. (If there is co course available, create or choose a course with only communication enabled, create a few example lecture/exercise posts and then enable messaging for the course).

1. Roll out this branch
2. For the test course, within the messages tab, there should be new channels for exercises/lectures
3. Channel messages should be visible on exercise/lecture pages
4. Courses, that do not have both communication and messaging enabled should still look as before.

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/main/resources/config/liquibase/changelog/20230601154000_changelog.xml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            FROM exercise e
+                     LEFT JOIN conversation c ON e.id = c.exercise_id
+                     LEFT JOIN course c2 on e.course_id = c2.id
+            WHERE c.exercise_id IS NULL
+              AND c2.info_sharing_config = 1
+              AND (c2.end_date IS NULL OR c2.end_date > now())
+              AND NOT EXISTS (
+                SELECT 1
+                FROM conversation
+                WHERE name = CONCAT('lecture-', LOWER(REPLACE(LEFT(e.title, 10), ' ', '-')), '-', e.id)
```
Comment (rriyaldhi) [inline src/main/resources/config/liquibase/changelog/20230601154000_changelog.xml:53]: why don't use the e.id = conversation.exercise_id condition here to check if an exercise already has a channel or not?

Comment (pellumbbaboci) [reply src/main/resources/config/liquibase/changelog/20230601154000_changelog.xml]: This check is here to make sure that the create channel does not conflict with any other channel name within the course. So if there is channel with same name the new channel will not be created.

**Review by MaximilianAnzinger — APPROVED** [review]
Tested on TS1 during testing session. Looks good to me.

**Review by tobias-lippert — APPROVED** [review]
Migration seems to work, no data seems to be lost and no errors occurred.

File: src/main/resources/config/liquibase/changelog/20230601154000_changelog.xml [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.18.xsd">
+    <changeSet author="baboci" id="20230601154000">
+        <comment>Create new channels for existing lecture/exercise and migrate posts of lecture/exercise to respective channel</comment>
+        <comment>Step 1: Create channels for lectures that do not have a channel and update posts</comment>
+        <sql>
+            INSERT INTO conversation (course_id, lecture_id, name, description,discriminator, is_public, is_announcement, creation_date)
+            SELECT l.course_id, l.id, CONCAT('lecture-',LOWER(REPLACE(LEFT(l.title, 10), ' ', '-')), '-', l.id), CONCAT('Description of lecture: ', l.title), 'C', true, false, NOW()
```
[the remark this answers is not in the document]
Comment (lennart-keller) [reply src/main/resources/config/liquibase/changelog/20230601154000_changelog.xml]: You are right, we removed it from the migration script

**Review by TheZoker — APPROVED** [review]

**Pull request merged**